### DISCUSSION TITLE
Aggregation Relationship

### DISCUSSION THREAD

**Discussion by walterconcert** [cimug-org/CIMTool#129] [General]
Hello everyone!
Is it expected that Aggregation relationship between classes doesn't appear in the Project Model view? Only association and generalization?
I have an .eap from customer that has some aggregations. As I prefer to work with CIMTool to build the profiles, it has becoming a problem for me.

**Comment by tviegut** [comment]
> relationship

Walter,

If I understand your question right...are you looking for something such as the following in the CIM?

![image](https://github.com/user-attachments/assets/0afc21a5-3553-4af9-9042-44737d6ae424)

**Reply by walterconcert** [reply]
Yes! This kind of representation, but for Aggregation relationship, not association, like your example.

**Reply by walterconcert** [reply]
Check this example for the Class DCTopologicalNode. It has an aggregation relationship with DCTopologicalIsland (EA). 
It doesn't appear in CIMTool, as you can see in the right side of the printscreen.
![image](https://github.com/user-attachments/assets/2fb60898-0652-4bc1-8be7-e31672e89f90)

**Reply by tviegut** [reply]
@walterconcert , I believe I've resolved the issue.  Below screenshots show both a Composite and an Aggregation respectively with the corresponding test UML (in red) at the bottom.  

My simple request (if you don't mind) is to formally create an issue to track the defect (https://github.com/cimug-org/CIMTool/issues) and add your screenshots to document the issues.  Also, please provide specific information with respect to whether your using an XMI for your schema in CIMTool or if your importing the .eap project file and using it directly in it.

Thanks in advance.  Depending on urgency I can either do a patch release or include it in the 2.1.0 release planned in the next handful of weeks.

Thanks for using CIMTool and for catching the defect.  

Todd

**Composition:**
![image](https://github.com/user-attachments/assets/76648cc8-76fc-47e5-b13a-28663f6f734c)

**Aggregation**:
![image](https://github.com/user-attachments/assets/7109c66c-ad09-4a61-80ae-8aef7a806dc0)

![image](https://github.com/user-attachments/assets/a26bf60f-3dfa-4d83-8ada-dd6416e27de9)  
![image](https://github.com/user-attachments/assets/eef0cfc8-f0e4-465d-ab87-79f6b61a60f6)

**Comment by tviegut** [comment]
@walterconcert , I just went ahead and created an issue off of this discussion thread.  So ignore my prior comments other than to add whether your using `.XMI` or `.eap` as your schema.  Just add your comments to https://github.com/cimug-org/CIMTool/issues/131

**Comment by walterconcert** [comment]
Thanks a lot, @tviegut. I addded the comment as you required in #131 .
I have to deliver the profiles by end of August to the customer. I can edit the profiles using EA (personally I prefer to use CIMTool), so it is not so urgent, but it would be great having the fix in CIMTool.
